routes/transaction.go

```diff
@@ -2547,6 +2547,277 @@ func (fes *APIServer) TransferDAOCoin(ww http.ResponseWriter, req *http.Request)
+	TransactionFees      []TransactionFee `safeForLogging:"true"`
+}
+
+// CancelDAOCoinLimitOrder Constructs a transaction that cancels an existing DAO coin limit order with the specified
+// order id
+func (fes *APIServer) CancelDAOCoinLimitOrder(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := DAOCoinLimitOrderWithCancelOrderIDRequest{}
+
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprintf("CancelDAOCoinLimitOrder: Problem parsing request body: %v", err),
+		)
+		return
+	}
+
+	cancelOrderID, err := decodeBlockHashFromHex(requestData.CancelOrderID)
+	if err != nil {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprintf("CancelDAOCoinLimitOrder: CancelOrderID param is not a valid order id: %v", err),
+		)
+		return
+	}
+
+	res, err := fes.validateAndCreateDAOCoinLimitOrderResponse(
+		requestData.TransactorPublicKeyBase58Check,
+		"",
+		"",
+		nil,
+		nil,
+		0,
+		cancelOrderID,
+		requestData.MinFeeRateNanosPerKB,
+		requestData.TransactionFees,
+	)
+
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("CancelDAOCoinLimitOrder: %v", err))
+		return
+	}
+
+	if err = json.NewEncoder(ww).Encode(res); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("CancelDAOCoinLimitOrder: Problem encoding response as JSON: %v", err))
+		return
+	}
+}
+
+func (fes *APIServer) validateAndCreateDAOCoinLimitOrderResponse(
+	transactorPublicKeyBase58Check string,
+	buyingDAOCoinCreatorPublicKeyBase58CheckOrUsername string,
+	sellingDAOCoinCreatorPublicKeyBase58CheckOrUsername string,
+	scaledExchangeRateCoinsToSellPerCoinToBuy *uint256.Int,
+	quantityToFillInBaseUnits *uint256.Int,
+	operationType lib.DAOCoinLimitOrderOperationType,
+	cancelOrderId *lib.BlockHash,
+	minFeeRateNanosPerKB uint64,
+	transactionFees []TransactionFee,
+) (*DAOCoinLimitOrderResponse, error) {
+
+	if transactorPublicKeyBase58Check == "" {
+		return nil, fmt.Errorf("must provide a TransactorPublicKeyBase58Check")
+	}
+
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		return nil, fmt.Errorf("problem fetching utxoView: %v", err)
+	}
+
+	// Decode the transactor public key
+	transactorPublicKeyBytes, _, err := fes.GetPubKeyAndProfileEntryForUsernameOrPublicKeyBase58Check(
+		transactorPublicKeyBase58Check,
+		utxoView,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("problem decoding transactor public key %s: %v",
+			transactorPublicKeyBase58Check, err)
+	}
+
+	// Compute the additional transaction fees as specified by the request body and the node-level fees.
+	additionalOutputs, err := fes.getTransactionFee(
+		lib.TxnTypeDAOCoinTransfer,
+		transactorPublicKeyBytes,
+		transactionFees,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("specified transactionFees are invalid: %v", err)
+	}
+
+	buyingCoinPublicKey := lib.ZeroPublicKey.ToBytes()
+	sellingCoinPublicKey := lib.ZeroPublicKey.ToBytes()
```

Comment:
should we initialize these to empty byte slices instead so we can exclude them from txn construction? Lib.NewPublicKey returns nil if the number of bytes isn't 33
```suggestion
	var buyingCoinPublicKey []byte
	var sellingCoinPublicKey []byte
```

Reply:
Ah, that's a good point. We'll want the public keys to be nil for order cancels. But we'll also want them to default to ZeroPublicKey when creating an order that's buying or selling DESO. I'll do a small refactor to account for this.